WooCommerce/Classes/Tools/Connectivity/DefaultConnectivityObserver.swift

```diff
@@ -8,16 +8,23 @@ final class DefaultConnectivityObserver: ConnectivityObserver {
     private let networkMonitor: NWPathMonitor
     private let observingQueue: DispatchQueue = .global(qos: .background)
 
-    var isConnectivityAvailable: Bool {
-        if case .reachable = connectivityStatus(from: networkMonitor.currentPath) {
-            return true
-        }
-        return false
+    @Published private(set) var isConnectivityAvailable: Bool = false
+    @Published private var connectivityStatus: ConnectivityStatus = .unknown
```

Comment:
AFAIK `@Published` wasn't created to just interact with SwiftUI views. The only limit of the property wrapper is that it's class constrained, so in cases where we need to work with a `struct`, `CurrentValueSubject` should be used instead. 

In our case here, I prefer `@Published` because I can explicitly use `private(set)` to make sure that the variable cannot be updated from outside. With `CurrentValueSubject`, I'll have to use type-eraser to make it an `AnyPublisher` to avoid changes from outside; and publishers are harder to test (requires testing async code).